### PR TITLE
[feature](time) using timev2 type to replace the old time type.

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.19 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17169590203 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.11 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17167321099 Bytes

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.03 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          71 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          35 seconds loaded 861443392 Bytes, about 23 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17162446504 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.34 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17168406428 Bytes

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.56 seconds
 stream load tsv:          504 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17160489970 Bytes

**Review by zclllyybb — APPROVED** [review]
LGTM

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.75 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17160473956 Bytes

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.02 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17169283633 Bytes

File: be/src/util/date_func.cpp [hunk 1]
```diff
@@ -118,13 +118,17 @@ int32_t time_to_buffer_from_double(double time, char* buffer) {
 
 int32_t timev2_to_buffer_from_double(double time, char* buffer, int scale) {
     static int pow10[7] = {1, 10, 100, 1000, 10000, 100000, 1000000};
+    const static int64_t max_time = (int64_t)3020399 * 1000 * 1000;
```
Comment (HappenLee) [inline be/src/util/date_func.cpp:121]: The value better be a CONST_VALUE, only define one time

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**